### DISCUSSION TITLE
Is there a stable hash code available for a SyntaxTree instance

### DISCUSSION THREAD

**Discussion by SeanFarrow-RSK** [dotnet/roslyn#69202] [Q&A]
Good morning,

I am trying to determine the equality of two SyntaxTree instances within an IEqualityComparer implementation.

I am aware I can use the IsEquivalentTo method to determine whether the two SyntaxTrees are equivalent, but can not find a way to compute the hashcode for a SyntaxTree.

Is there a stable GetHashCode implementation exposed for SyntaxTrees?

Any help apreciated.

Kind regards
Sean.

**Comment by jasonmalinowski** [comment]
@SeanFarrow-RSK What do you need it for? If you just need a GetHashCode implementation to be approximately correct, something as simple as the length of the tree might suffice. Usually IsEquivalentTo we don't use for comparing different trees of unknown origins, but rather to understand if certain changes or edits had other impacts.

**Comment by SeanFarrow-RSK** [comment]
@jasonmalinowski 
I am trying to compare two PropertyDeclarationSyntax instances and was getting stuck comparing the syntax trees.

Is there any better way of doing this.

This is for testing using FluentAssertions within a source generator.

**Reply by CyrusNajmabadi** [reply]
You can use the XxxEquivalentYyy methods on SyntaxNode and SyntaxFactory.

**Reply by jasonmalinowski** [reply]
@SeanFarrow-RSK: but why do you need a hash code for them? Just because you're needing to write an IEqualityComparer or something?
